src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt

```diff
@@ -705,34 +692,38 @@ class OffenderResourceIntTest_dischargeToHospital : ResourceTest() {
   }
 
   private fun createOffenderBooking() {
-    OffenderBuilder().withBooking(
-      OffenderBookingBuilder(
-        prisonId = "SYI",
-        cellLocation = "SYI-A-1-1",
-        programProfiles = listOf(OffenderProgramProfileBuilder()),
-        payStatuses = listOf(OffenderPayStatusBuilder()),
-        noPayPeriods = listOf(OffenderNoPayPeriodBuilder()),
-      ),
-    )
-      .save(testDataContext)
-      .also {
-        offenderNo = it.offenderNo
-        bookingId = it.bookingId
-      }
+    builder.build {
+      offenderNo = offender {
+        bookingId = booking(prisonId = "SYI", cellLocation = "SYI-A-1-1") { }.bookingId
+      }.offenderNo
+    }
   }
 
   private fun createOffenderWithNoBooking() {
-    OffenderBuilder(bookingBuilders = arrayOf()).save(testDataContext).also {
-      offenderNo = it.offenderNo
-      bookingId = it.bookingId
+    builder.build {
+      offenderNo = offender {
+      }.offenderNo
```

Comment:
I'm not sure about this syntax - it means if there are any children defaulted in the Offender DSL you are overriding them. Might be better to do `offenderNo = offender().offenderNo`?

Reply:
Not sure I know what you mean? `offender` returns an object which has a field `offenderNo` in